src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,110 @@
 }
 
-type DefaultedProps = RequiredProps & DefaultProps;
-
-@props(Props, { strict: true })
-export class Dropdown extends React.Component<Dropdown.Props> {
-  static defaultProps: DefaultProps = {
+export class Dropdown<OptionType> extends React.Component<
+  Dropdown.Props<OptionType>
+> {
+  static defaultProps: Partial<Dropdown.Props<{}>> = {
     delimiter: ",",
     size: "medium",
-    disabled: false,
-    searchable: false,
-    clearable: false,
-    multi: false,
+    isDisabled: false,
+    isSearchable: false,
     flat: false,
-    autoBlur: true,
-    simpleValue: true,
-    groupByKey: "optionGroup",
-    optionGroupRenderer: defaultOptionGroupRenderer,
-    menuRenderer: defaultMenuRenderer,
-    menuPosition: "bottom"
-  };
-
-  componentDidMount() {
-    this.logWarnings();
-  }
-
-  logWarnings = () => {
-    if (this.props.children) {
-      warn("You're passing children. Not expected behaviour");
-    }
+    blurInputOnSelect: true,
+    menuPlacement: "bottom",
+    components: {}
   };
 
-  valueToOption = (
-    value: Dropdown.Value | undefined,
-    options: SelectNS.Options
-  ) => {
-    if (t.String.is(value) || t.Number.is(value)) {
-      const { multi, delimiter } = this.props as DefaultedProps;
-      if (multi) {
-        const values = String(value).split(delimiter);
-        return compact(
-          values.map(v => find(options, { value: v }) || undefined)
-        );
-      }
-      return find(options, { value }) || undefined;
-    }
-    return value;
-  };
-
-  sortOptionsByGroup = (options: SelectNS.Options) => {
-    const { groupByKey } = this.props as DefaultedProps;
-    return sortBy(
-      options,
-      option =>
-        option[groupByKey]
-          ? findIndex(options, o => option[groupByKey] === o[groupByKey])
-          : -1
-    );
+  defaultComponents: Dropdown.Props<OptionType>["components"] = {
+    IndicatorSeparator: () => null
   };
 
   getCustomClassNames() {
-    const { size, flat, clearable, menuPosition } = this
-      .props as DefaultedProps;
+    const { size, flat, isSearchable } = this.props;
     return cx({
       "is-medium": size === "medium",
       "is-small": size === "small",
       "is-flat": flat,
-      "is-clearable": clearable,
-      "menu-position-top": menuPosition === "top"
+      "is-multi": this.isMulti(),
+      "is-searchable": isSearchable
     });
   }
 
-  _onChange = (_value?: Dropdown.Value | null) => {
-    const value = _value === "" || isEmptyArray(_value) ? null : _value;
-    return this.props.onChange(value);
-  };
-
-  onInputKeyDown: SelectNS.OnInputKeyDownHandler = e => {
-    if (e.keyCode === 38 || e.keyCode === 40) {
-      if (isEmptyArray(this.props.options)) {
-        e.preventDefault();
-      }
+  isMulti = (): boolean => {
+    switch (this.props.type) {
+      case "multi":
+      case "multi-clearable":
+        return true;
+      case "single":
+      case "single-clearable":
+        return false;
```

Comment:
with the `noFallthroughCasesInSwitch` option we are using in the default tsconfig in scriptoni, this would fail